pp.c

```diff
@@ -7253,6 +7253,76 @@ PP(pp_unweaken)
     RETURN;
 }
 
+PP(pp_blessed)
+{
+    dSP;
+    SV *arg = TOPs;
+    SV *rv;
+
+    SvGETMAGIC(arg);
+
+    if(!SvROK(arg) || !SvOBJECT((rv = SvRV(arg)))) {
+        SETs(&PL_sv_undef);
+        RETURN;
+    }
+
+    if((PL_op->op_private & OPpTRUEBOOL) ||
+            ((PL_op->op_private & OPpMAYBE_TRUEBOOL) && (block_gimme() == G_VOID))) {
```

Comment:
What is the special casing here for void context? Is blessed in void context even useful?

Reply:
`OPpMAYBE_TRUEBOOL` means that it's the final statement in a sub, so we can only optimise it away if it's in void context, since we don't know what the sub return value is used for.

Reply:
Huh, the more I think about it the more I'm not quite sure. I copied that from the similar code in `pp_ref`. Perhaps that needs more staring at.

Reply:
Ah - I see @ilmari has already answered that :)

Reply:
I'm still confused but I can live with that.